libs/wingsdk/src/target-sim/redis.inflight.ts

```diff
@@ -96,4 +90,42 @@ export class Redis implements IRedisClient, ISimulatorResourceInstance {
       throw new Error("Redis server not initialized");
     }
   }
+
+  public async set(key: string, value: string): Promise<void> {
+    let redis = await this.rawClient();
+    await redis.set(key, value);
+  }
+
+  public async get(key: string): Promise<string> {
```

Comment:
I'd say that the (wing) return value here needs to be a `str?`, I think this translates to a JSII `Promise<string | null>`.
(to handle the case of getting a non-existent key)

Reply:
hmm, im not sure how we pass back an optional from the SDK. @Chriscbr do we do this somewhere else? JSII doesnt seem to be happy with me when I try to do <string | null>

Reply:
I think JSII has some equivalent of an optional value, the jsii_importer has code to support this, see `optional_type_to_wing_type`.

Reply:
Maybe a `Promise<string> | Promise<undefined>`?

Reply:
Was able to make this work by patching JSii basically copied what @Chriscbr did for his `try_get` apis on bucket (it just had not finished PR) https://github.com/winglang/wing/pull/1622